packages/webdriverio/src/commands/mobile/pinch.ts

```diff
@@ -0,0 +1,53 @@
+import { getBrowserObject } from '@wdio/utils'
+import type { PinchAndZoomOptions } from '../../types.js'
+import { calculateAndroidPinchAndZoomSpeed, validatePinchAndZoomOptions } from '../../utils/mobile.js'
+
+/**
+ *
+ * Performs a pinch gesture on the given element on the screen.
+ *
+ * :::info
+ *
+ * Pinching is done based on native mobile gestures. It is only supported for the following drivers:
+ * - [appium-uiautomator2-driver](https://github.com/appium/appium-uiautomator2-driver/blob/master/docs/android-mobile-gestures.md#mobile-pinchclosegesture) for Android
+ * - [appium-xcuitest-driver](https://appium.github.io/appium-xcuitest-driver/latest/reference/execute-methods/#mobile-pinch) for iOS
+ *
+ * :::
+ *
+ * <example>
+    :pinch.js
+    it('should demonstrate a pinch on Google maps', async () => {
+        const mapsElement = $('//*[@resource-id="com.google.android.apps.maps:id/map_frame"]')
+        // Pinch with the default duration scale
+        await mapsElement.pinch()
+        // Pinch with a custom duration and scale
+        await mapsElement.pinch({ duration: 4000, scale: 0.9 })
+    })
+ * </example>
+ *
+ * @alias element.pinch
+ * @param {PinchOptions=} options           pinch options (optional)
+ * @param {number=}       options.duration  The duration in millisecond of how fast the pinch should be executed, minimal is 500 ms and max is 10000 ms. The default is 1500 ms (1.5 seconds) (optional)
+ * @param {number=}       options.scale     The scale of how big the pinch should be according to the screen. Valid values must be float numbers in range 0..1, where 1.0 is 100% (optional)
+ */
+export async function pinch(
+    this: WebdriverIO.Element,
+    options: Partial<PinchAndZoomOptions> = {}
+) {
+    const browser = getBrowserObject(this)
+    const { duration, scale } = validatePinchAndZoomOptions({ browser, gesture: 'pinch', options })
```

Comment:
Can we add a check for both commands and verify that we are in the right environment, e.g. mobile and (not sure if required) native context.

Reply:
@christian-bromann 
The validation is done in the validatePinchZoomOptions method

Reply:
How about we create a generic method that verifies whether the browser is in a certain environment, e.g. 

```ts
enum ENVIRONMENTS {
  WEB_ONLY = 1,
  MOBILE_NATIVE_ONLY
  MOBILE_WEB_ONLY
}

verifyEnvironmentForCommandExecution (supportedEnvironments: ENVIRONMENTS[]) {
  // ...
}
```

So we can put this in front of all commands that have an environment restriction. I think keeping this as part of the command implementation rather than as part of a sub function makes it more clear where this command can be executed when looking at its implementation.

Reply:
Great idea, but this might have more impact than only this PR 😅

Reply:
Happy to split this into a separate PR, however I still think we should keep the environment check as part of the command implementation for visibility purposes.

Reply:
Hmm, not sure if I understand you correctly. Do you mean you want the check in one of the first lines, but in a generic method? Or do you want a HOC for all components?

Reply:
Fixed it, thanks